src/plugins/data_source_management/public/plugin.ts

```diff
@@ -303,10 +319,85 @@ export class DataSourceManagementPlugin
     };
     setRenderAssociatedObjectsDetailsFlyout(renderAssociatedObjectsDetailsFlyout);
 
+    const getDirectQuerySyncFromUrl = (): boolean | null => {
+      const parsedHash = parseUrlHash(window.location.href);
+      if (parsedHash && parsedHash.query) {
+        const directQuerySync = parsedHash.query.directQuerySync;
+        if (directQuerySync === 'true') return true;
```

Comment:
qq: instead of having a value it's either the url has it or not. is it possible that it could have neither of these values? might it be easier to assume the presence fo the para is true. and you can explicitly set it to false

also we should do something similar to how the `?embed` does it 

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/dashboard/public/application/components/dashboard_top_nav/dashboard_top_nav.tsx#L62